src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -5615,6 +5723,93 @@ def update_identity(self, mc: ManagedCluster) -> ManagedCluster:
             mc.identity = identity
         return mc
 
+    def update_azure_keyvault_secrets_provider_addon_profile(
```

Comment:
The implementation of this function basically follows [this part](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/custom.py#L2715-L2731) in file `custom.py`.